core/src/main/scala/org/apache/spark/resource/ResourceUtils.scala

```diff
@@ -47,7 +48,24 @@ private[spark] case class ResourceRequest(
     discoveryScript: Option[String],
     vendor: Option[String])
 
-private[spark] case class ResourceRequirement(resourceName: String, amount: Int)
+/**
+ * Case class that represents a user's resource requirement as given by configuration
```

Comment:
nit: Instead of mentioning `a user's resource requirement`, I would prefer `a task` or `an application`